### PR TITLE
PWN-3250 - Show error when incorrect seed phrase is entered

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket (Mandatory)

https://p2pvalidator.atlassian.net/browse/PWN-3250

## Description of Work (Mandatory)

- disable button
- make framelayout red when validation fails
- (optional) cleanup SecretKeyFragment according to code style

## Screenshots (Optional)
Screen Shot 2022-04-18 at 11 34 40 am

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/restore/ui/keys/SecretKeyFragment.kt [hunk 1]
```diff
@@ -42,18 +46,20 @@ class SecretKeyFragment :
...
     private val analyticsInteractor: ScreensAnalyticsInteractor by inject()
 
-    private val phraseAdapter: SecretPhraseAdapter by lazy {
-        SecretPhraseAdapter {
-            presenter.setNewKeys(it)
-            clearError()
-        }
+    private val phraseAdapter: SecretPhraseAdapter by unsafeLazy {
+        SecretPhraseAdapter(
+            onSeedPhraseChanged = { keys ->
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/restore/ui/keys/SecretKeyFragment.kt:58]: i couldn't guess on what action this callback invokes. I would need to go inside inside adapter.

File: app/src/main/java/org/p2p/wallet/common/ui/widget/ProgressButton.kt [hunk 2]
```diff
@@ -4,83 +4,114 @@ import android.content.Context
...
         context.theme.resolveAttribute(android.R.attr.selectableItemBackground, outValue, true)
+
         foreground = getDrawable(context, outValue.resourceId)
         clipToOutline = true
 
         isClickable = true
         isFocusable = true
 
-        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.ProgressButton)
+        context.obtainStyledAttributes(attrs, R.styleable.ProgressButton).use { typedArray ->
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/common/ui/widget/ProgressButton.kt:38]: ofc

File: app/src/main/java/org/p2p/wallet/restore/BackupModule.kt [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
-import org.koin.dsl.module
 
 object BackupModule : InjectionModule {
 
     override fun create() = module {
-        factory { SecretKeyInteractor(get(), get(), get(), get(), get(), get(), get()) }
-        factory { SecretKeyPresenter(androidContext().resources, get(), get()) } bind SecretKeyContract.Presenter::class
-        factory { (secretKeys: List<SecretKey>) ->
+        factory {
+            SecretKeyInteractor(get(), get(), get(), get(), get(), get(), get())
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/restore/BackupModule.kt:17]: named args?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/restore/BackupModule.kt:17]: hmm, agreed!

File: app/src/main/java/org/p2p/wallet/restore/ui/keys/SecretKeyFragment.kt [hunk 4]
```diff
@@ -99,9 +111,28 @@ class SecretKeyFragment :
         binding.restoreButton.isEnabled = isEnabled
     }
 
-    override fun showError(messageRes: Int) {
-        binding.errorTextView.setText(messageRes)
-        binding.messageTextView.isVisible = false
+    override fun showError(messageRes: Int) = with(binding) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/restore/ui/keys/SecretKeyFragment.kt:114]: @StringRes

**Review by kabramovp2p — APPROVED** [review]

**Review by davran1001 — APPROVED** [review]

**Pull request merged**